dnsserv.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"github.com/twtiger/toy-dns-nameserver/reqhandler"
+	"net"
+)
+
+func getUDPAddr() *net.UDPAddr {
+	return &net.UDPAddr{
+		IP:   net.ParseIP("127.0.0.1"),
+		Port: 53,
+	}
+}
+
+func run() error {
+	for {
+		udpConn, err := net.ListenUDP("udp", getUDPAddr())
+		if err != nil {
+			return err
+		}
+		go reqhandler.HandleUDPConnection(udpConn)
+		udpConn.Close()
```

Comment:
I believe putting udpConn.Close right after the go routine causes Close to be called before the request handler has been called. "Normally when we invoke a function our program will execute all the statements in a function and then return to the next line following the invocation. With a goroutine we return immediately to the next line and don't wait for the function to complete."

Ref: https://www.golang-book.com/books/intro/10